api/serializers.py

```diff
@@ -42,6 +44,7 @@ class Meta:
         ]
         read_only_fields = [
             "id",
+            "type",
```

Comment:
issue: You're putting the `type` as read-only. This needs to be set when generating a mask from the add-on.